File: newBot/services/video_editor_service.py
from .backend_client import BackendClient


class VideoEditorService:
    def __init__(self) -> None:
        self.client = BackendClient()

    def register_video_editor(
        self,
        user_id: int,
        full_name: str,
        city: str,
        inn: str,
        phone: str,
        business_type: str,
        bik: str,
        account: str,
        bank_name: str,
        bank_ks: str,
        bank_details: str,
    ) -> dict:
        payload = {
            "userId": user_id,
            "fullName": full_name,
            "city": city,
            "inn": inn,
            "phone": phone,
            "businessType": business_type,
            "bik": bik,
            "account": account,
            "bankName": bank_name,
            "bankKs": bank_ks,
            "bankDetails": bank_details,
        }
        return self.client.post("video-editor/bot", payload)

    def approve_video_editor(self, user_id: int) -> bool:
        self.client.put(f"video-editor/bot/{user_id}", {"approved": True})
        return True

    def sign_video_editor_contract(self, user_id: int) -> None:
        self.client.put(f"video-editor/bot/{user_id}", {"contractSigned": True})

    def get_video_editor_profile(self, user_id: int) -> dict:
        return self.client.get(f"video-editor/bot/{user_id}")

    def remove_video_editor(self, user_id: int) -> bool:
        self.client.delete(f"video-editor/bot/{user_id}")
        return True

File: backend/src/modules/video-editor/video-editor.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from '@/src/core/prisma/prisma.service';
import { plainToInstance } from 'class-transformer';
import { CreateVideoEditorDto } from './dto/create-video-editor.dto';
import { UpdateVideoEditorDto } from './dto/update-video-editor.dto';
import { VideoEditorResponseDto } from './dto/video-editor-response.dto';

@Injectable()
export class VideoEditorService {
    constructor(private readonly prismaService: PrismaService) { }

    async create(data: CreateVideoEditorDto): Promise<VideoEditorResponseDto> {
        const ve = await this.prismaService.videoEditor.create({
            data: {
                user: { connect: { id: data.userId } },
                fullName: data.fullName,
                city: data.city,
                inn: data.inn,
                phone: data.phone,
                businessType: data.businessType,
                bik: data.bik,
                account: data.account,
                bankName: data.bankName,
                bankKs: data.bankKs,
                bankDetails: data.bankDetails,
                approved: data.approved,
                contractSigned: data.contractSigned,
            },
            include: { user: true },
        });
        return plainToInstance(VideoEditorResponseDto, { ...ve.user, ...ve });
    }

    async findAll(): Promise<VideoEditorResponseDto[]> {
        const list = await this.prismaService.videoEditor.findMany();
        return list.map((l) => plainToInstance(VideoEditorResponseDto, l));
    }

    async findOne(id: number): Promise<VideoEditorResponseDto> {
        const ve = await this.prismaService.videoEditor.findUnique({ where: { id } });
        if (!ve) {
            throw new NotFoundException('VideoEditor not found');
        }
        return plainToInstance(VideoEditorResponseDto, ve);
    }

    async update(id: number, data: UpdateVideoEditorDto): Promise<VideoEditorResponseDto> {
        const ve = await this.prismaService.videoEditor.update({ where: { userId: id }, data });
        return plainToInstance(VideoEditorResponseDto, ve);
    }

    async remove(id: number): Promise<VideoEditorResponseDto> {
        const ve = await this.prismaService.videoEditor.delete({ where: { id } });
        return plainToInstance(VideoEditorResponseDto, ve);
    }
}